File: tests/fixtures/sample.ts
import { readFileSync } from "fs";
import * as path from "path";

const MAX_RETRIES: number = 3;
export const DEFAULT_NAME: string = "world";

export interface Speakable {
    name: string;
    sound: string;
    speak(): string;
    describe(): string;
}

export class Animal implements Speakable {
    public name: string;
    public sound: string;

    constructor(name: string, sound: string) {
        this.name = name;
        this.sound = sound;
    }

    public speak(): string {
        return this.sound;
    }

    public describe(): string {
        return "I am " + this.name;
    }
}

export class Dog extends Animal {
    public tricks: string[];

    constructor(name: string) {
        super(name, "woof");
        this.tricks = [];
    }

    public learnTrick(trick: string): void {
        this.tricks.push(trick);
    }

    public perform(trick: string): boolean {
        if (this.tricks.includes(trick)) {
            return true;
        }
        return false;
    }
}

export function greet(name: string): string {
    return "Hello, " + name;
}

export function fibonacci(n: number): number {
    if (n <= 1) {
        return n;
    }
    let a: number = 0;
    let b: number = 1;
    let i: number = 2;
    while (i <= n) {
        const temp: number = a + b;
        a = b;
        b = temp;
        i = i + 1;
    }
    return b;
}

export function findMax(numbers: number[]): number | null {
    if (numbers.length === 0) {
        return null;
    }
    let best: number = numbers[0];
    for (const x of numbers) {
        if (x > best) {
            best = x;
        }
    }
    return best;
}

export function countWords(text: string): Map<string, number> {
    const counts: Map<string, number> = new Map();
    for (const word of text.split(" ")) {
        const current = counts.get(word) ?? 0;
        counts.set(word, current + 1);
    }
    return counts;
}

export enum Direction {
    North,
    South,
    East,
    West,
}

File: tests/fixtures/sample.py
"""Sample Python file for round-trip testing."""
import os
from typing import List, Optional, Dict

MAX_RETRIES: int = 3
DEFAULT_NAME: str = "world"


class Animal:
    """Base class for animals."""

    name: str
    sound: str = "..."

    def __init__(self, name: str, sound: str) -> None:
        self.name = name
        self.sound = sound

    def speak(self) -> str:
        return self.sound

    def describe(self) -> str:
        return f"I am {self.name}"


class Dog(Animal):
    """A dog that can fetch."""

    tricks: List[str]

    def __init__(self, name: str) -> None:
        self.name = name
        self.sound = "woof"
        self.tricks = []

    def learn_trick(self, trick: str) -> None:
        self.tricks.append(trick)

    def perform(self, trick: str) -> bool:
        if trick in self.tricks:
            return True
        return False


def greet(name: str) -> str:
    return "Hello, " + name


def fibonacci(n: int) -> int:
    if n <= 1:
        return n
    a: int = 0
    b: int = 1
    i: int = 2
    while i <= n:
        temp: int = a + b
        a = b
        b = temp
        i = i + 1
    return b


def find_max(numbers: List[int]) -> Optional[int]:
    if not numbers:
        return None
    best: int = numbers[0]
    for x in numbers:
        if x > best:
            best = x
    return best


def count_words(text: str) -> Dict[str, int]:
    counts: Dict[str, int] = {}
    for word in text.split():
        if word in counts:
            counts[word] = counts[word] + 1
        else:
            counts[word] = 1
    return counts
